vulntotal/tests/test_gitlab.py

```diff
@@ -32,17 +32,35 @@ def test_generate_package_advisory_url(self):
         expected_file = self.get_test_loc("package_advisory_url-expected.json", must_exist=False)
         util_tests.check_results_against_json(results, expected_file)
 
-    def test_parse_html_advisory(self):
+    @mock.patch("vulntotal.datasources.gitlab.fetch_yaml")
+    def test_parse_interesting_advisories(self, mock_fetch_yaml):
+        # Mock the yaml file responses
         advisory_folder = (
             Path(__file__)
             .resolve()
-            .parent.joinpath("test_data/gitlab/temp_vulntotal_gitlab_datasource")
-        )
-        results = [
-            adv.to_dict()
-            for adv in gitlab.parse_interesting_advisories(
-                advisory_folder, PackageURL("generic", "namespace", "test", "0.1.1"), False
+            .parent.joinpath(
+                "test_data/gitlab/temp_vulntotal_gitlab_datasource/gemnasium-db-master-pypi-Jinja2/pypi/Jinja2"
             )
+        )
+        yaml_files = []
+        sorted_files = sorted(advisory_folder.iterdir(), key=lambda x: x.name)
+        for file in sorted_files:
+            if file.suffix == ".yml":
+                with open(file, "r") as f:
+                    yaml_files.append(f.read())
+
+        mock_fetch_yaml.side_effect = yaml_files
+
+        purl = PackageURL("generic", "namespace", "test", "0.1.1")
+
+        yml_files = [
+            {"name": "CVE-2014-1402.yml", "path": "path/to/CVE-2014-1402.yml"},
+            {"name": "CVE-2016-10745.yml", "path": "path/to/CVE-2016-10745.yml"},
```

Comment:
@michaelehab  restore `vulntotal/tests/test_data/gitlab/temp_vulntotal_gitlab_datasource/gemnasium-db-master-pypi-Jinja2/pypi/Jinja2/CVE-2019-8341.yml`. This test file is important for testing false positives. VulnTotal should not report the vulnerability in this file since version `0.1.1` is not in the affected range for `CVE-2019-8341`.

```suggestion
            {"name": "CVE-2016-10745.yml", "path": "path/to/CVE-2016-10745.yml"},
            {"name": "CVE-2019-8341.yml", "path": "path/to/CVE-2019-8341.yml"},
```

Reply:
Thank you for catching that! I apologize for missing the inclusion of the CVE-2019-8341 file. I've re-added it to the yml_files and restored the test file the test function. I ran the tests, and everything is passing now. Thanks again for your attention to detail!